### PR TITLE
Fix/need protection for feedback replies

### PR DESCRIPTION
**Pull request by teodorrosusn**
[AB#97877](https://dev.azure.com/plentymarkets/0af57df9-8662-4901-bb97-8e88b07ff0c9/_workitems/edit/97877)

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "version": "4.0.10",
```
Comment (Dominik2809) [inline plugin.json:2]: Do not change yet

File: resources/lang/de/Debug.properties [hunk 2]
```diff
@@ -1,2 +1,4 @@
 filterResponse=Filter Antwort
 FeedbackService_itemDoesNotExistError=Der angeforderte Artikel existiert nicht
+determineVisibilityResponse = Determine visibility response
```
Comment (Dominik2809) [inline resources/lang/de/Debug.properties:3]: Where is this used?

File: resources/lang/en/Debug.properties [hunk 3]
```diff
@@ -1,2 +1,4 @@
 filterResponse=Filter response
 FeedbackService_itemDoesNotExistError=The requested item does not exist
+determineVisibilityResponse = Determine visibility response
```
Comment (Dominik2809) [inline resources/lang/en/Debug.properties:3]: Where is this used?

File: src/Services/FeedbackService.php [hunk 4]
```diff
@@ -14,6 +14,8 @@
 use Plenty\Modules\Item\Attribute\Contracts\AttributeValueNameRepositoryContract;
 use Plenty\Modules\Item\Item\Contracts\ItemRepositoryContract;
 use Plenty\Modules\Item\Variation\Contracts\VariationRepositoryContract;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
```
Comment (Dominik2809) [inline src/Services/FeedbackService.php:17]: Is this usable in plugins?

File: src/Services/FeedbackService.php [hunk 5]
```diff
@@ -644,4 +658,28 @@ private function getUserIdFromOrder($order)
...
+     * @param int $feedbackId
+     *
+     * @return bool
+     */
+    private function feedbackExists(int $feedbackId): bool
+    {
+        try {
+            /** @var Feedback $feedback */
+            $feedback = $this->feedbackRepository->getFeedback($feedbackId);
+        } catch (ModelNotFoundException) {}
```
Comment (Dominik2809) [inline src/Services/FeedbackService.php:672]: Not assigning a variable is valid for PHP 8 but to be safe, assign one

**Review by Dominik2809 — CHANGES_REQUESTED** [review]

File: resources/lang/de/Debug.properties (hunk 2 above)
Comment (teodorrosusn) [reply resources/lang/de/Debug.properties:3]: Was used for log debugging purposes. I've removed it now.

File: plugin.json (hunk 1 above)
Comment (teodorrosusn) [reply plugin.json]: This was also used for testing. Changed back to 4.0.10.

File: resources/lang/en/Debug.properties (hunk 3 above)
Comment (teodorrosusn) [reply resources/lang/en/Debug.properties:3]: Was used for log debugging purposes. I've removed it now.

File: src/Services/FeedbackService.php (hunk 5 above)
Comment (teodorrosusn) [reply src/Services/FeedbackService.php:672]: Done.

File: src/Services/FeedbackService.php (hunk 4 above)
Comment (teodorrosusn) [reply src/Services/FeedbackService.php:17]: Right now yes, in the "feedbackExists" method.

File: src/Services/FeedbackService.php [hunk 6]
```diff
@@ -644,4 +658,28 @@ private function getUserIdFromOrder($order)
...
+    private function feedbackExists(int $feedbackId): bool
+    {
+        try {
+            /** @var Feedback $feedback */
+            $feedback = $this->feedbackRepository->getFeedback($feedbackId);
+        } catch (ModelNotFoundException $exception) {}
+
+        $this->getLogger(__METHOD__)->debug('Feedback::Debug.feedbackExistsResult', [
+            'expectedFeedbackId' => $feedbackId,
+            'obtainedFeedbackId' => $feedback->id
```
Comment (Dominik2809) [inline src/Services/FeedbackService.php:676]: isn't this a null pointer in case an exception was caught before this?

**Review by Dominik2809 — CHANGES_REQUESTED** [review]

File: src/Services/FeedbackService.php (hunk 4 above)
Comment (Dominik2809) [reply src/Services/FeedbackService.php:17]: I meant if the ModelNotFoundException is in the plugin interface to be used inside plugins

**Review by Dominik2809 — APPROVED** [review]

**Pull request merged**